src/node/internal/internal_http_server.ts

```diff
@@ -0,0 +1,708 @@
+
+      if (contentLength >= 0) {
+        // @ts-expect-error TS2304 Fix this once global types are correct.
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument,@typescript-eslint/no-unsafe-call
+        body = body.pipeThrough(new FixedLengthStream(contentLength));
+      }
+    }
+
+    return new Response(body, {
+      status: statusCode,
+      statusText,
+      headers,
+    });
+  }
+
+  #dataFromDataWrittenEvent({
+    index,
+    entry: { data, encoding },
+  }: DataWrittenEvent): Buffer | Uint8Array {
+    if (index === 0) {
+      if (typeof data !== 'string') {
+        console.error('First chunk should be string, not sure what happened.');
+        throw new ERR_INVALID_ARG_TYPE(
+          'packet.data',
+          ['string', 'Buffer', 'Uint8Array'],
+          data
+        );
+      }
+      // The first X bytes are header material, so we remove it.
+      data = data.slice(this.writtenHeaderBytes);
+    }
+
+    if (typeof data === 'string') {
+      if (
+        encoding === undefined ||
+        encoding === 'utf8' ||
+        encoding === 'utf-8'
+      ) {
+        data = this.#encoder.encode(data);
+      } else {
+        data = Buffer.from(data, encoding ?? undefined);
+      }
+    }
+
+    return data ?? Buffer.alloc(0);
+  }
+
+  assignSocket(_socket: Socket): void {
+    // We don't plan to support this method, since our
+    // implementation is based on fetch and not Node.js sockets.
+    throw new ERR_METHOD_NOT_IMPLEMENTED('assignSocket');
+  }
+
+  detachSocket(_socket: Socket): void {
+    // We don't plan to support this method, since our
+    // implementation is based on fetch and not Node.js sockets.
+    throw new ERR_METHOD_NOT_IMPLEMENTED('detachSocket');
+  }
+
+  writeContinue(_cb: VoidFunction): void {
+    // There is no path forward to support this with fetch or kj.
+    throw new ERR_METHOD_NOT_IMPLEMENTED('writeContinue');
+  }
+
+  // Ref: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73275
+  // @ts-expect-error TS2416 This is a bug with @types/node.
+  writeProcessing(_cb: VoidFunction): void {
+    // There is no path forward to support this with fetch or kj.
+    throw new ERR_METHOD_NOT_IMPLEMENTED('writeProcessing');
+  }
+
+  writeEarlyHints(_hints: unknown, _cb: VoidFunction): void {
+    // There is no path forward to support this with fetch or kj.
+    throw new ERR_METHOD_NOT_IMPLEMENTED('writeEarlyHints');
+  }
+
+  override _implicitHeader(): void {
+    this.writeHead(this.statusCode);
+  }
+
+  writeHead(
+    statusCode: number,
+    reason?: string | OutgoingHttpHeaders | OutgoingHttpHeader[],
+    obj?: OutgoingHttpHeaders | OutgoingHttpHeader[]
+  ): this {
+    if (this._header) {
+      throw new ERR_HTTP_HEADERS_SENT('write');
+    }
+
+    const originalStatusCode = statusCode;
+
+    statusCode |= 0;
+    if (statusCode < 100 || statusCode > 999) {
```

Comment:
Since we know 1xx codes aren't supported at all, let's just filter them out here to save us some effort later.